### DISCUSSION TITLE
Is there a convenient way to batch process a folder of structures?

### DISCUSSION THREAD

**Discussion by alainCollas** [jmaglic/MoloVol#120] [Q&A] [answered]
It seems the CLI is suited to automate workflows yet is there an elegant way for processing whole folders? Thanks in advance!

**Comment by jmaglic** [comment, marked as answer]
Hi! There is currently no in-built way to process multiple input files at once. However, the CLI provides enough flexibility that it is possible to do so using a simple script. Let me demonstrate using bash.

I called the script file `multifile` (you can find the code at the bottom of this post). It can be called from the command line using `bash multifile arg1 arg2`. There are two input arguments:

`arg1` is the command that you use to run MoloVol from the command line with your specific input parameters, excluding the `--file-structure` argument. This could, for instance, be:

`"/Applications/MoloVol.app/Contents/MacOS/MoloVol --radius 1.2 --grid 0.2"`

For a guide on how to use the CLI, refer to the manual or the (hopefully up-to-date) [wiki page](https://github.com/molovol/MoloVol/wiki/Command-Line-Interface).

Note three things: 1) The whole command must be contained in double quotation marks, so that the script recognises it as a single input argument, 2) `/Applications/MoloVol.app/Contents/MacOS/MoloVol` is the path to the executable under macOS. This path may be different on your platform.

`arg2` is a list of the input files you want to process. You could list them individually, but your command line window should provide wildcard characters that make this much simpler. Suppose your input files are contained inside the folder `./inputs/`, `arg2` could be:

`"./inputs/*"`

Note again the need for double quotation marks so that the input files are recognised as part of a list. To process only .xyz files you can add a small modification:

`"./inputs/*.xyz"`

Long story short, to use the bash script from the command line, run the following command from the folder containing the bash script:

```console
bash multifile "/Applications/MoloVol.app/Contents/MacOS/MoloVol --radius 1.2 --grid 0.2" "./inputs/*"
```

This will sequentially process all files contained inside the subdirectory `./inputs/` using the CLI, and the same specified input parameters.

Because there are an endless variations of specific use-cases, we can't hope to provide in-build functionalities to cover all of these. But I hope that the CLI, in combination with a scripting language, is flexible enough to provide a robust tool set.

### `multifile`
```bash
#!/bin/bash

MV_COMMAND=$1
INPUTFILES=$2

for file in $INPUTFILES
do
  echo "`$MV_COMMAND -fs $file`"
done
```

**Reply by alainCollas** [reply]
Many thanks for this suggestion. I got it operational after replacing the switches with the verbose. Helps a lot!
